lightning/src/ln/channelmanager.rs

```diff
@@ -2905,16 +2920,30 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 						// event being update_fulfill_htlc).
 						if let Err(e) = self.chain_monitor.update_channel(prev_outpoint, preimage_update) {
 							log_error!(self.logger, "Critical error: failed to update channel monitor with preimage {:?}: {:?}",
-							           payment_preimage, e);
+									   payment_preimage, e);
```

Comment:
I'm trying to parse why we keep `claimed_htlc = true` in every error case... Is it the case that, for all non-duplicates, claiming is guaranteed to "succeed" due to falling back to on-chain in the worst case?

Reply:
I added a comment, let me know if its clearer now.